diff --git a/vulkano/src/command_buffer/synced/base.rs b/vulkano/src/command_buffer/synced/base.rs
index 4cd60b86..4d679bc8 100644
--- a/vulkano/src/command_buffer/synced/base.rs
+++ b/vulkano/src/command_buffer/synced/base.rs
@@ -546,6 +546,10 @@ impl SyncCommandBufferBuilder {
                     resource_index,
                 };
 
+                for builder_key in self.resources.keys() {
+                    &key == builder_key;
+                }
+
                 // Note that the call to `entry()` will lock the mutex, so we can't keep it locked
                 // throughout the function.
                 match self.resources.entry(key) {
